src/shim/index.js

```diff
@@ -122,6 +122,18 @@ const commands = {
 function read() {
     let line = system.stdin.readLine();
     if (line) {
+        if (line === 'NOOP') {
+            //if sender has done a noop,
+            //it means sender is not busy
+            //so lets give phantom more time
+            setTimeout(function() {
```

Comment:
Interesting. I like that you tested it so I feel confident that it will be better. Also it makes sense why you need the boolean above now because of this timeout. Otherwise it would fill the queue.